external-api/src/middleware/check-api-key/index.ts

```diff
@@ -0,0 +1,40 @@
+import dotenv from 'dotenv';
+import { Request, Response, NextFunction } from 'express';
+
+dotenv.config();
+
+const { API_KEY } = process.env;
+
+/**
+ * Check that the x-api-key header is valid
+ * @param {Object} req Request object
+ * @param {Object} res Response object
+ * @param {String} next Callback function name
+ */
+export const checkApiKey = (req: Request, res: Response, next: NextFunction) => {
+  const {
+    headers: { 'x-api-key': xApiKey },
+  } = req;
+
+  /**
+   * x-api-key is not provided.
+   * Reject the request.
+   */
+  if (!xApiKey) {
+    return res.status(400).send();
```

Comment:
should it still not be 401 unauthorized since not provided?

Reply:
We discussed in slack, I think this makes sense because it's genuinely a bad request

Reply:
Just jumping here, it needs to be 401. 400 is a bad request that will be applicable if the body is invalid. 
We need to keep it consistent with everything else

Reply:
Updated 👍